consensus/merkle_test.go

```diff
@@ -45,36 +45,66 @@ func TestElementAccumulatorEncoding(t *testing.T) {
 
 func TestStorageProofRoot(t *testing.T) {
     leafHash := types.Hash256{0x01, 0x02, 0x03}
-    proof := []types.Hash256{
-        {0x04, 0x05, 0x06},
-        {0x07, 0x08, 0x09},
+    validProof := []types.Hash256{
+        {0x0A, 0x0B, 0x0C},
+        {0x0D, 0x0E, 0x0F},
+        {0x10, 0x11, 0x12},
+        {0x13, 0x14, 0x15},
     }
+    longProof := append(validProof, types.Hash256{0x16, 0x17, 0x18})
+    shortProof := validProof[:2]
+
+    validWantRoot := types.Hash256{0xa8, 0x9d, 0xbb, 0xb5, 0x45, 0xaa, 0x4b, 0x46, 0x69, 0x62, 0x30, 0xd1, 0x04, 0x07, 0x6f, 0x06, 0xb5, 0x7a, 0x6a, 0xb0, 0x8b, 0x23, 0x41, 0xc3, 0xd0, 0x12, 0xbd, 0xc1, 0x3e, 0x23, 0xeb, 0x35}
```

Comment:
you can make this more compact with `UnmarshalText`:
```go
var validWantRoot types.Hash256
validWantRoot.UnmarshalText([]byte(`h:a89dbbb545aa4b46696230d104076f06b57a6ab08b2341c3d012bdc13e23eb35`))
```

Reply:
Would it be right to also add an error message to report that the hash does not match the expected format?

```go
var validWantRoot types.Hash256
    err := validWantRoot.UnmarshalText([]byte("a89dbbb545aa4b46696230d104076f06b57a6ab08b2341c3d012bdc13e23eb35"))
    if err != nil {
        t.Fatalf("Error deserializing the textual representation of the storage proof root hash: %v", err)
    }

Reply:
In this case it's not really necessary. If the `UnmarshalText` fails, `validWantRoot` will be all zeros and the test will still fail.